### PR TITLE
Feature/eip1 4286 EMS Integration API - Initial setup

### PR DESCRIPTION
**Pull request by spalamthodi**

### CODE REVIEW

**Review by inderjitvirdi — APPROVED** [review]

File: src/test/resources/certificates/dwp/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+This directory contains a self signed certificate that is used by the integration tests in place of the DWP client certificate.
+
+* `client-cert.pem` - the client certificate file
+* `private-key.pem` - the private key used to generate the certificate
```
Comment (gurpiarbassi) [inline src/test/resources/certificates/dwp/README.md:4]: will this service talk to DWP?

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+spring:
+  datasource:
+    url: jdbc:mysql://${MYSQL_HOST}:${MYSQL_PORT}/ems_integration_application
```
Comment (gurpiarbassi) [inline src/main/resources/application.yml:3]: will this service use mysql or dynamodb? still hasnt been decided I think

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+cloud:
+  aws:
+    region:
+      auto: true
+      use-default-aws-region-chain: true
+    credentials:
+      use-default-aws-credentials-chain: true
+
+sqs:
+  accepted-proxy-application-queue-name: ${SQS_ACCEPTED_PROXY_APPLICATION_QUEUE_NAME}
```
Comment (gurpiarbassi) [inline src/main/resources/application.yml:19]: is all this relevant?

File: build.gradle.kts [hunk 4]
```diff
@@ -1,33 +1,193 @@
...
+    implementation("io.projectreactor.netty:reactor-netty-http")
+
+    // spring security
+    implementation("org.springframework.boot:spring-boot-starter-security")
+    implementation("org.springframework.boot:spring-boot-starter-oauth2-resource-server")
+
+    implementation("io.awspring.cloud:spring-cloud-starter-aws-messaging")
+    implementation("org.springframework.boot:spring-boot-starter-data-jpa")
+    implementation("org.liquibase:liquibase-core")
+    implementation("org.hibernate:hibernate-envers")
```
Comment (gurpiarbassi) [inline build.gradle.kts:72]: I think for this new service we ought to depend on spring-data-envers which will being in hibernate-envers transitively

File: README.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
+# ems-integration-api
+
+Spring Boot microservice that :
+
+- Consumes approved Postal and Proxy application SQS message from EROP and saved into the database
```
Comment (gurpiarbassi) [inline README.md:5]: relevant?

File: src/main/java/com/amazonaws/http/AWSRequestSigningApacheInterceptor.java [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+import java.util.Map;
+import java.util.TreeMap;
+
+import static org.apache.http.protocol.HttpCoreContext.HTTP_TARGET_HOST;
+
+/**
+ * An {@link HttpRequestInterceptor} that signs requests using any AWS {@link Signer}
+ * and {@link AWSCredentialsProvider}.
+ */
+public class AWSRequestSigningApacheInterceptor implements HttpRequestInterceptor {
```
Comment (gurpiarbassi) [inline src/main/java/com/amazonaws/http/AWSRequestSigningApacheInterceptor.java:54]: do we need this for this app?
I think this is only used if we have comms with IER that require requests to be signed

**Pull request merged**